src/util/AbstractMemory/Memory.h

```diff
@@ -0,0 +1,114 @@
+// Copyright 2023, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Andre Schlegel (July of 2023,
+// schlegea@informatik.uni-freiburg.de)
+
+#pragma once
+
+#include <cmath>
+#include <concepts>
+#include <cstddef>
+#include <string>
+#include <type_traits>
+
+#include "util/AbstractMemory/CalculationUtil.h"
+#include "util/Exception.h"
+
+namespace ad_utility {
+
+/*
+An abstract class, that represents an amount of memory.
+Note: Literals for easier usage of the class were defined after under the class
+definition.
+*/
+class Memory {
+  // Because of `sizeof` we know, that any size of memory, in bytes, can be
+  // encoded as a `size_t` in cpp.
+  size_t memoryInBytes_;
+
+ public:
+  // Very simple custom constructor.
+  Memory(const size_t& amountOfMemoryInBytes = 0)
+      : memoryInBytes_{amountOfMemoryInBytes} {}
+
+  // Default assignment operator.
+  Memory& operator=(const Memory&) = default;
+  Memory& operator=(Memory&&) = default;
+
+  // Custom assignment operator, so that we can assign user literals.
+  Memory& operator=(const size_t& amountOfMemoryInBytes);
+  Memory& operator=(size_t&& amountOfMemoryInBytes);
```

Comment:
We shouldn't have implicit conversion from numbers, but you should always have to call
`Memory::Bytes(4213)` or something like this.

Reply:
Do you mean explicit setter, or a static member function?

Reply:
This would be a static factory function.
For example you can write `std::chrono::miliseconds(5)` instead of `5ms` in `std::chrono` and I would suggest something else here.